src/main/java/org/wildfly/security/auth/client/AuthenticationContextConfigurationClient.java

```diff
@@ -134,7 +134,7 @@ public AuthenticationConfiguration getAuthenticationConfiguration(URI uri, Authe
             final IdentityCredentials privateCredentials = securityIdentity.getPrivateCredentials();
             final IdentityCredentials publicCredentials = securityIdentity.getPublicCredentials();
             // private overrides public
-            configuration = configuration.useForwardedIdentity(null).usePrincipal(securityIdentity.getPrincipal()).useCredentials(publicCredentials.with(privateCredentials));
+            configuration = configuration.useForwardedIdentity(securityIdentity.getPrincipal() == AnonymousPrincipal.getInstance() ? forwardSecurityDomain : null).usePrincipal(securityIdentity.getPrincipal()).useCredentials(publicCredentials.with(privateCredentials));
```

Comment:
This is probably wrong and it is a workaround to overcome the following issue.

In https://github.com/wildfly/wildfly-core/blob/master/remoting/subsystem/src/main/java/org/jboss/as/remoting/RemoteOutboundConnectionService.java#L111, when obtaining a ```AuthenticationConfiguration``` the ```SecurityDomain``` set to forward identities is lost, causing propagation to fail due to the lack of the current security principal.

I'm not sure about the correct place to fix that. If changing ```RemoteOutboundConnectionService``` or providing a specific method on ```AuthenticationContextConfigurationClient``` to avoid setting the ```SecurityDomain``` to null in order to cover ```RemoteOutboundConnectionService``` requirements.

@dmlloyd What do you think ?